app/controllers/lists_controller.rb

```diff
@@ -18,9 +18,28 @@ def edit
   end
 
   def create
-    @list = List.new
+    @list = List.new(list_params)
+
+    if @list.valid?
+      @list.save
+
+      redirect_to @list
+    else
+      flash[:notice] = @list.errors.full_messages if @list.errors.count > 0
+
+      redirect_to lists_path
+    end
+  end
+
+  def update
+    @list = List.find(params[:id])
+
+    @list.update(list_params)
+    @list.name = list_params[:name]
+
+    @list.save if @list.valid?
```

Comment:
`.save` calls `.valid?` and returns false if the record is invalid

Reply:
Right! Thanks! Fixed in e41970e